### DISCUSSION TITLE
I'm making the mod of CK3 clothing. Most of the models are one-sided meshes. I import the exported model into blender again and find that there are many more vertices than originally. Is this normal? Or is there something wrong with my settings?

### DISCUSSION THREAD

**Discussion by LGGHC** [ross-g/io_pdx_mesh#66] [Q&A] [answered]
I'm making the mod of CK3 clothing. Most of the models are one-sided meshes. I import the exported model into blender again and find that there are many more vertices than originally. Is this normal? Or is there something wrong with my settings?
This may cause the clothing deformation model to not be read correctly by the game

**Comment by ross-g** [comment, marked as answer]
Hi, there is nothing wrong or unexpected here... "exporting" isn't the same as "saving" and you shouldn't expect that what you export (or re-import) will be the same as what you started with. Changes do happen on export that are necessary simply because 3D software and game engines treat the data differently, some changes are well known (eg game engines require triangle meshes, so I triangulate on export) but other changes are less obvious.

For example:
A simple plane in Blender, already triangulated. On the right you can see it's UVs, which are split in half.
How many vertices are there? 4
How many UVs are there? 6
In Blender the top-left and bottom-right vertices each have one position, but two UV coordinates.
![image](https://user-images.githubusercontent.com/9258768/161401044-57b3d4d0-b1d8-4d92-af63-25910c496fd7.png)

In most game engines, this isn't possible... you can't have a vertex with variable amounts of data (ie 1 position, 2 UVs etc). So on export, at any point where there is a UV seam more vertices are written out into the mesh file than existed in Blender. Exporting that example above will produce a mesh with six vertices in game and if you re-import that it will show a different number of vertices than you had originally.

This edge-splitting happens on:

- UV seams (ie split UV vertices)
- Normal changes (ie sharp edge shading)
- Material boundaries (ie changing material across faces)
